testing/regress/ecl/mongodb-simple.ecl

```diff
@@ -0,0 +1,96 @@
+dataset(analyticsrec) findTransactionsByID(INTEGER id) := Embed(mongodb : user(username), password(password), server(server), database('sample_analytics'),  collection('transactions'))
+    find_one({account_id: $id});
+ENDEMBED;
+
+// Returns the document with a transaction count greater than or equal to the number passed in.
+dataset(analyticsrec) findTransactionsByCount(INTEGER trans_count) := EMBED(mongodb : user(username), password(password), server(server), database('sample_analytics'),  collection('transactions'))
+    find_one({transaction_count: {$gte: $trans_count}});
+ENDEMBED;
+
+// Returns the document with a transaction count less than the number passed in.
+dataset(analyticsrec) findTransactionsBelowCount(INTEGER trans_count) := EMBED(mongodb : user(username), password(password), server(server), database('sample_analytics'),  collection('transactions'))
+    find_one({transaction_count: {$lt: $trans_count}});
+ENDEMBED;
+
+// Returns the document matching the id that is passed in.
+dataset(airbnbrec) findABByID(STRING id) := EMBED(mongodb : user(username), password(password), server(server), database('sample_airbnb'),  collection('listingsAndReviews'))
+    find_one({_id: $id});
+ENDEMBED;
+
+// Returns a dataset. Finds all the documents where the beds and accomodates fields are higher than the argument values.
+dataset(airbnbrec) findABBeds(INTEGER beds, INTEGER accommodates) := EMBED(mongodb : user(username), password(password), server(server), database('sample_airbnb'),  collection('listingsAndReviews'))
+    find_one({beds: {$gte: $beds}, accommodates: {$gte: $accommodates}});
+ENDEMBED;
+
+// Returns the document matching the url that is passed in.
+dataset(airbnbrec) findABByListing(STRING url) := EMBED(mongodb : user(username), password(password), server(server), database('sample_airbnb'),  collection('listingsAndReviews'))
+    find_one({listing_url: $url});
+ENDEMBED;
+
+// Returns the document matching the id that is passed in.
+dataset(feesrec) findFeeByID(STRING id) := EMBED(mongodb : user(username), password(password), server(server), database('sample_airbnb'),  collection('listingsAndReviews'))
+    find_one({_id: $id});
+ENDEMBED;
+
+// Returns a dataset. Finds all the documents where the beds and accomodates fields are higher than the argument values.
+dataset(feesrec) findFeebyBeds(INTEGER beds, INTEGER accommodates) := EMBED(mongodb : user(username), password(password), server(server), database('sample_airbnb'),  collection('listingsAndReviews'))
+    find_one({beds: {$gte: $beds}, accommodates: {$gte: $accommodates}});
+ENDEMBED;
+
+// Returns the document matching the url that is passed in.
+dataset(feesrec) findFeeByListing(STRING url) := EMBED(mongodb : user(username), password(password), server(server), database('sample_airbnb'),  collection('listingsAndReviews'))
+    find_one({listing_url: $url});
+ENDEMBED;
+
+SEQUENTIAL
+(
+    OUTPUT(findTransactionsByID(716662), NAMED('FindTransactionByID'));
```

Comment:
ideally the output of these commands should be reproducible - is there sample/generated dataset(s) required for these commands to execute as expected?

Reply:
Yes, there is an option in MongoDB atlas to load a "sample dataset" which includes multiple collections. I have loaded it on two different accounts and they were the same datasets, but there is nothing stopping MongoDB from changing the sample datasets that they include in the future.

Reply:
Fair enough. So are these commands expecting one of those datasets?
Regardless of the source of the data, we should ensure we document how to load the data these commands expect.

Reply:
Yes, these commands are expecting to see the sample datasets from MongoDB, but the way that MongoDB operates is if you specify a collection that does not exist it will create it on the cluster. In order to load every dataset that the commands are expecting there is only a single step which I could create some documentation for. I believe it is trivial and MongoDB already offers documentation for it, but I could create some additional documentation that would make it even simpler since I have done it a few times.

Reply:
Got it!
Ok so, sample_airbnb.listingsAndReviews and  sample_airbnb.transactions are made available on the cluster automatically.
I think a quick comment here will go a long way.

Reply:
Yes, definitely! I will add a comment so there is no confusion.